distribution/helm/vector-aggregator/templates/_helpers.tpl

```diff
@@ -1,18 +1,55 @@
 {{/* vim: set filetype=mustache: */}}
 
 {{/*
-Resolve effective service ports to use.
+Internal template to render service ports depending on whether service is a headless service or not. Use
+either 'vector-aggregator.servicePorts' or 'vector-aggregator.headlessServicePorts' as entry points.
 */}}
-{{- define "vector-aggregator.servicePorts" -}}
-{{- if .Values.vectorSource.enabled }}
+{{- define "vector-aggregator.internalServicePorts" -}}
+{{- $headless := index . 0 -}}
+{{- $values := index . 1 -}}
+{{- if $values.vectorSource.enabled }}
 - name: vector
-  port: {{ .Values.vectorSource.listenPort }}
+{{- if and $values.vectorSource.nodePort (not $headless) (eq "NodePort" $values.service.type) }}
```

Comment:
```suggestion
{{- if and $values.vectorSource.nodePort (not $headless) }}
```

We should set the `nodePort` if it's defined, regardless of what's the service type is. This will work better for `LoadBalancer` type, that also respects the `nodePort` setting.